modules/minimp3/resource_importer_mp3.cpp

```diff
@@ -112,7 +113,13 @@ Error ResourceImporterMP3::import(ResourceUID::ID p_source_id, const String &p_s
 	mp3_stream->set_beat_count(beat_count);
 	mp3_stream->set_bar_beats(bar_beats);
 
-	return ResourceSaver::save(mp3_stream, p_save_path + ".mp3str");
+	int flags = ResourceSaver::FLAG_COMPRESS;
+#ifdef TOOLS_ENABLED
+	if (!EDITOR_GET("filesystem/on_save/compress_binary_resources")) {
+		flags &= ~ResourceSaver::FLAG_COMPRESS;
+	}
+#endif
+	return ResourceSaver::save(mp3_stream, p_save_path + ".mp3str", flags);
```

Comment:
The check is inverted here as we're outside `editor/`, so we can only conditionally check for the editor setting. When outside the editor, compression is always enabled, which matches the default editor behavior. (I'm not sure how importers can run outside the editor, if it's even possible in the first place, so it might not matter either way.)